Retail data improvements

vinibackbase opened the pull request:
Product Data
- Adding Savings Account;
- Removing unnecessary data;

Tested on http://dawn-sonic-31-editorial.backbase.test:8080/gateway/retail-banking-demo-wc3/login#/
![Screen Shot 2019-08-26 at 13 20 12](https://user-images.githubusercontent.com/47349985/63687235-4be16d80-c804-11e9-96a0-3b04fb4a46ac.png)
<img width="842" alt="Screen Shot 2019-08-26 at 13 20 52" src="https://user-images.githubusercontent.com/47349985/63687253-5c91e380-c804-11e9-883a-3f658250a8de.png">

richard-blank commented on src/main/java/com/backbase/ct/bbfuel/setup/AccessControlSetup.java:
```diff
@@ -200,8 +200,14 @@ protected void ingestDataGroupArrangementsForServiceAgreement(String internalSer
+                    arrangementIds = this.productSummaryConfigurator.ingestRetailArrangements(
+                            externalLegalEntityId);
+                } else {
+                    arrangementIds = this.productSummaryConfigurator.ingestArrangements(
```
If it is not retail what kind of arrangements are these? I think it would be better to make random/static data decision on something else. Business arrangements could be loaded statically as well

vinibackbase replied:
It shouldn't have any difference, to be honest. But, in order not to break other teams who are relying on EUR arrangements and the random product seed.

richard-blank replied:
if you keep the product group seed as default way of data generation you will not break anything

vinibackbase replied:
but i won't have as much control of the data because the way that the code is structured in the productConfigurator. The json file that was present in this PR is just a first iteration. Product is providing more input so the data doesn't gibberish like you saw.

richard-blank replied:
i don't see the added value of having the same named accounts for all retail users. If you have a small environment with only a couple of them i would suggest to refactor the product seed group mechanism and make it more flexible on that point. The solution as proposed now cannot be the default.

richard-blank commented on src/main/resources/data.properties:
```diff
@@ -8,6 +8,7 @@ products.json=data/products.json
 product.group.seed.json=data/product-group-seed.json
 service.agreements.json=data/service-agreements.json
 legal.entities.with.users.json=data/legal-entities-with-users.json
+arrangements.data.json=data/retail-arrangements.json
```
the name is logical but the 'random' arrangements are seeded from `product.group.seed.json`. It would be better to have some name similarity so it is easy to detect their connection

vinibackbase replied:
fixed.

richard-blank commented on src/main/resources/data/retail-arrangements.json:
```diff
@@ -0,0 +1,56 @@
+  {
+    "BBAN": "123456",
+    "BIC": "bICOWgEl",
+    "accountHolderAddressLine1": "Pt0Ed",
```
For being static data it seems very random. Could you give it more real life names?

vinibackbase replied:
This is just a starting point, I am waiting on more data from product. Just making a PR to get ahead.

richard-blank commented on src/main/java/com/backbase/ct/bbfuel/data/CommonConstants.java:
```diff
@@ -81,4 +81,7 @@ private CommonConstants() {
     public static final String PROPERTY_TRANSACTIONS_DATA_JSON = "transactions.data.json";
+
+    // Arrangements
+    public static final String PROPERTY_PRODUCTS_DATA_JSON = "retail-products-data.json";
```
property name is `product.data.json`

richard-blank commented on src/main/java/com/backbase/ct/bbfuel/configurator/ProductSummaryConfigurator.java:
```diff
@@ -28,12 +29,31 @@
+        List<ArrangementsPostRequestBody> arrangements = synchronizedList(new ArrayList<>());
+        List<ArrangementId> arrangementIds = synchronizedList(new ArrayList<>());
+
+        arrangements.addAll(reader.load(externalLegalEntityId));
```
side note: although it is a small file it is not efficient to load it for each retail customer.

vinibackbase commented:
Adding evidence for new proposed approach
![Screen Shot 2019-09-03 at 16 04 37](https://user-images.githubusercontent.com/47349985/64184552-e6c4f200-ce6b-11e9-9039-ee504d0d587a.png)
<img width="841" alt="Screen Shot 2019-09-03 at 16 03 40" src="https://user-images.githubusercontent.com/47349985/64184559-ea587900-ce6b-11e9-9c28-53c284a6b952.png">

vinibackbase commented:
Reverted new changes and just added some tweaks to the Retail Banking product group did the magic.

![Screen Shot 2019-09-04 at 15 58 55](https://user-images.githubusercontent.com/47349985/64261726-40412580-cf2d-11e9-8fdd-bf45740d4501.png)
![Screen Shot 2019-09-04 at 15 59 04](https://user-images.githubusercontent.com/47349985/64261730-420ae900-cf2d-11e9-866b-dd10e1476042.png)
